### PR TITLE
History screen

### PR DESCRIPTION
**Pull request by matzapata**
## Overview

Create history tabs for aave

## Changelog

- Create graphql schema with apollo
- Create tabs for earning and borrow in the history screen and connect them only for bob chain

## Demo

image
image
image
image
image

## Notes

Pending one more testing round and maybe some cleanup

### CODE REVIEW

File: apps/frontend/src/app/3_organisms/BorrowHistory/components/AaveCloseWithDepositLoanFrame/AaveCloseWithDepositLoanFrame.tsx [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        type: NotificationType.warning,
+        title: t(translations.common.tables.actions.noDataToExport),
+        content: '',
+        dismissible: true,
+        id: nanoid(),
+      });
+      return [];
+    }
+
+    return normalizeUserRepayTransactions(data).map(item => ({
```
Comment (matzapata) [inline apps/frontend/src/app/3_organisms/BorrowHistory/components/AaveCloseWithDepositLoanFrame/AaveCloseWithDepositLoanFrame.tsx:125]: recheck and expand

File: apps/frontend/src/app/3_organisms/BorrowHistory/components/AaveCloseWithDepositLoanFrame/AaveCloseWithDepositLoanFrame.tsx [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+    pageSize,
+    page,
+    orderOptions,
+  );
+
+  useEffect(() => {
+    refetch();
+  }, [refetch, block]);
+
+  const [getBorrowHistory] = useUserRepayTransactionsLazyQuery({
```
Comment (matzapata) [inline apps/frontend/src/app/3_organisms/BorrowHistory/components/AaveCloseWithDepositLoanFrame/AaveCloseWithDepositLoanFrame.tsx]: rename `getBorrowHistory`

File: apps/frontend/src/app/3_organisms/BorrowHistory/components/AaveCloseWithDepositLoanFrame/AaveCloseWithDepositLoanFrame.tsx [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    client: bobAaveClient,
+  });
+
+  const generateRowTitle = useCallback(
+    (row: any) => (
+      <Paragraph size={ParagraphSize.small} className="text-left">
+        {t(translations.borrowHistory.transactionTypes.createLoan)}
```
Comment (matzapata) [inline apps/frontend/src/app/3_organisms/BorrowHistory/components/AaveCloseWithDepositLoanFrame/AaveCloseWithDepositLoanFrame.tsx]: wrong translation

File: apps/frontend/src/app/3_organisms/BorrowHistory/components/AaveCloseWithDepositLoanFrame/AaveCloseWithDepositLoanFrame.tsx [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+  const exportLocked = checkMaintenance(States.ZERO_EXPORT_CSV);
+
+  return (
+    <>
+      <div className="flex-row items-center gap-4 mb-7 flex justify-center lg:justify-start">
+        {children}
+        <div className="flex-row items-center ml-2 gap-4 hidden lg:inline-flex">
+          <ExportCSV
+            getData={exportData}
+            filename="New loans"
```
Comment (matzapata) [inline apps/frontend/src/app/3_organisms/BorrowHistory/components/AaveCloseWithDepositLoanFrame/AaveCloseWithDepositLoanFrame.tsx]: rename

File: apps/frontend/src/app/3_organisms/BorrowHistory/components/AaveCloseWithDepositLoanFrame/AaveCloseWithDepositLoanFrame.tsx [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+          setOrderOptions={setOrderOptions}
+          orderOptions={orderOptions}
+          columns={COLUMNS}
+          rows={data}
+          rowTitle={generateRowTitle}
+          isLoading={loading}
+          className="bg-gray-80 text-gray-10 lg:px-6 lg:py-4"
+          noData={t(translations.common.tables.noData)}
+          loadingData={t(translations.common.tables.loading)}
+          dataAttribute="new-loans-table"
```
Comment (matzapata) [inline apps/frontend/src/app/3_organisms/BorrowHistory/components/AaveCloseWithDepositLoanFrame/AaveCloseWithDepositLoanFrame.tsx]: rename

File: apps/frontend/src/app/3_organisms/BorrowHistory/components/AaveCloseWithDepositLoanFrame/AaveCloseWithDepositLoanFrame.tsx [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+        />
+        <Pagination
+          page={page}
+          className="lg:pb-6 mt-3 lg:mt-6 justify-center lg:justify-start"
+          onChange={onPageChange}
+          itemsPerPage={pageSize}
+          isNextButtonDisabled={isNextButtonDisabled}
+          dataAttribute="new-loans-pagination"
```
Comment (matzapata) [inline apps/frontend/src/app/3_organisms/BorrowHistory/components/AaveCloseWithDepositLoanFrame/AaveCloseWithDepositLoanFrame.tsx]: same

File: apps/frontend/src/app/3_organisms/BorrowHistory/components/AaveCloseWithDepositLoanFrame/AaveCloseWithDepositLoanFrame.types.tsx [hunk 7]
```diff
@@ -0,0 +1,13 @@
+export type NewBorrowHistoryItem = {
```
Comment (matzapata) [inline apps/frontend/src/app/3_organisms/BorrowHistory/components/AaveCloseWithDepositLoanFrame/AaveCloseWithDepositLoanFrame.types.tsx]: rename

File: apps/frontend/src/app/3_organisms/BorrowHistory/components/AaveLiquidationLoanFrame/AaveLiquidationLoanFrame.tsx [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+  }, [refetch, block]);
+
+  const [getLiquidationsHistory] = useUserLiquidationsTransactionsLazyQuery({
+    client: bobAaveClient,
+  });
+
+  const generateRowTitle = useCallback(
+    (row: any) => (
+      <Paragraph size={ParagraphSize.small} className="text-left">
+        {t(translations.borrowHistory.transactionTypes.createLoan)}
```
Comment (matzapata) [inline apps/frontend/src/app/3_organisms/BorrowHistory/components/AaveLiquidationLoanFrame/AaveLiquidationLoanFrame.tsx]: rename

File: apps/frontend/src/app/3_organisms/BorrowHistory/components/AaveLiquidationLoanFrame/AaveLiquidationLoanFrame.tsx [hunk 9]
```diff
@@ -0,0 +1,191 @@
...
+        type: NotificationType.warning,
+        title: t(translations.common.tables.actions.noDataToExport),
+        content: '',
+        dismissible: true,
+        id: nanoid(),
+      });
+      return [];
+    }
+
+    return normalizeUserLiquidationTransactions(data).map(item => ({
```
Comment (matzapata) [inline apps/frontend/src/app/3_organisms/BorrowHistory/components/AaveLiquidationLoanFrame/AaveLiquidationLoanFrame.tsx:125]: recheck

File: apps/frontend/src/app/3_organisms/BorrowHistory/components/AaveLiquidationLoanFrame/AaveLiquidationLoanFrame.tsx [hunk 10]
```diff
@@ -0,0 +1,191 @@
...
+  const exportLocked = checkMaintenance(States.ZERO_EXPORT_CSV);
+
+  return (
+    <>
+      <div className="flex-row items-center gap-4 mb-7 flex justify-center lg:justify-start">
+        {children}
+        <div className="flex-row items-center ml-2 gap-4 hidden lg:inline-flex">
+          <ExportCSV
+            getData={exportData}
+            filename="New loans"
```
Comment (matzapata) [inline apps/frontend/src/app/3_organisms/BorrowHistory/components/AaveLiquidationLoanFrame/AaveLiquidationLoanFrame.tsx]: rename

File: apps/frontend/src/app/3_organisms/BorrowHistory/components/AaveNewLoanFrame/AaveNewLoanHistoryFrame.utils.tsx [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+    return [
+      ...acc,
+      {
+        id: tx.id,
+        amount: Decimal.from(
+          formatUnits(tx.amount, tx.reserve.decimals),
+        ).toString(2),
+        borrowRate: Decimal.from(tx.borrowRate).div(RAY).toString(2),
+        borrowRateMode:
+          tx.borrowRateMode === BorrowRateMode.STABLE ? 'Stable' : 'Variable', // TODO: translation
```
Comment (matzapata) [inline apps/frontend/src/app/3_organisms/BorrowHistory/components/AaveNewLoanFrame/AaveNewLoanHistoryFrame.utils.tsx]: todo

File: apps/frontend/src/app/3_organisms/EarnHistoryFrame/components/AaveLendingHistoryFrame/AaveLendingHistoryFrame.constants.tsx [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+import { AmountRenderer } from '../../../../2_molecules/AmountRenderer/AmountRenderer';
+import { TransactionIdRenderer } from '../../../../2_molecules/TransactionIdRenderer/TransactionIdRenderer';
+import { getCurrentChain } from '../../../../../hooks/useChainStore';
+import { translations } from '../../../../../locales/i18n';
+import { dateFormat } from '../../../../../utils/helpers';
+import { LendingHistoryItem } from './AaveLendingHistoryFrame.types';
+
+export const COLUMNS_CONFIG = [
+  {
+    id: 'time',
```
Comment (matzapata) [inline apps/frontend/src/app/3_organisms/EarnHistoryFrame/components/AaveLendingHistoryFrame/AaveLendingHistoryFrame.constants.tsx]: timestamp

File: apps/frontend/src/app/3_organisms/EarnHistoryFrame/components/AaveLendingHistoryFrame/AaveLendingHistoryFrame.constants.tsx [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+    title: t(translations.common.tables.columnTitles.amount),
+    cellRenderer: (item: LendingHistoryItem) =>
+      item.amount ? (
+        <AmountRenderer
+          value={formatUnits(item.amount, item.reserve.decimals)}
+          precision={2}
+          showRoundingPrefix
+        />
+      ) : (
+        <span>N/A</span>
```
Comment (matzapata) [inline apps/frontend/src/app/3_organisms/EarnHistoryFrame/components/AaveLendingHistoryFrame/AaveLendingHistoryFrame.constants.tsx]: join toghether

File: apps/frontend/src/app/3_organisms/EarnHistoryFrame/components/AaveLendingHistoryFrame/AaveLendingHistoryFrame.tsx [hunk 14]
```diff
@@ -0,0 +1,133 @@
...
+  );
+
+  const transactions = useMemo(
+    () => (data?.userTransactions as LendingHistoryItem[]) || [],
+    [data?.userTransactions],
+  );
+
+  useEffect(() => {
+    console.log('orderOptions', orderOptions);
+  }, [orderOptions]);
```
Comment (matzapata) [inline apps/frontend/src/app/3_organisms/EarnHistoryFrame/components/AaveLendingHistoryFrame/AaveLendingHistoryFrame.tsx]: remove

File: apps/frontend/src/constants/aave.ts [hunk 15]
```diff
@@ -1,9 +1,14 @@
+import { Decimal } from '@sovryn/utils';
+
 import { IS_MAINNET } from '../config/chains';
 
 import { decimalic } from '../utils/math';
 
+export const RAY = Decimal.from(10).pow(27);
```
Comment (matzapata) [inline apps/frontend/src/constants/aave.ts]: move to utils/math.ts

File: schema.graphql
Comment (matzapata) [inline schema.graphql:1]: remove

File: apps/frontend/src/app/3_organisms/BorrowHistory/components/AaveLiquidationLoanFrame/AaveLiquidationLoanFrame.tsx [hunk 16]
```diff
@@ -0,0 +1,193 @@
...
+  useEffect(() => {
+    refetch();
+  }, [refetch, block]);
+
+  const [getLiquidationsHistory] = useUserLiquidationsTransactionsLazyQuery({
+    client: bobAaveClient,
+  });
+
+  const generateRowTitle = useCallback(
+    (row: any) => (
```
Comment (juandahl) [inline apps/frontend/src/app/3_organisms/BorrowHistory/components/AaveLiquidationLoanFrame/AaveLiquidationLoanFrame.tsx:77]: replace any by the type

**Review by juandahl — APPROVED** [review]
LGTM

**Pull request merged**